src/components/spreadsheet/table-wrapper.js

```diff
@@ -0,0 +1,686 @@
+                            params: {
+                                setEditingData: setEditingData,
+                                setIsValidatingData: setIsValidatingData,
+                                startEditing: startEditing,
+                            },
+                        };
+                    } else if (editingData?.id === params.data.id) {
+                        return {
+                            component: ReferenceLineCellRenderer,
+                        };
+                    } else {
+                        return {
+                            component: EditableCellRenderer,
+                            params: {
+                                setEditingData: setEditingData,
+                                equipmentType:
+                                    TABLES_DEFINITION_INDEXES.get(tabIndex)
+                                        .modifiableEquipmentType,
+                            },
+                        };
+                    }
+                },
+            });
+        },
+        [editingData?.id, startEditing, tabIndex]
+    );
+
+    const generateTableColumns = useCallback(
+        (tabIndex) => {
+            const generatedTableColumns = TABLES_DEFINITION_INDEXES.get(
+                tabIndex
+            )
+                .columns.filter((c) => {
+                    return selectedColumnsNames.has(c.id);
+                })
+                .map((column) => enrichColumn(column));
+
+            function sortByIndex(a, b) {
+                if (reorderedTableDefinitionIndexes) {
+                    if (
+                        reorderedTableDefinitionIndexes.indexOf(a.id) <
+                        reorderedTableDefinitionIndexes.indexOf(b.id)
+                    ) {
+                        return -1;
+                    }
+                    if (
+                        reorderedTableDefinitionIndexes.indexOf(a.id) >
+                        reorderedTableDefinitionIndexes.indexOf(b.id)
+                    )
+                        return 1;
+                }
+                return 0;
+            }
+            generatedTableColumns.sort(sortByIndex);
+
+            if (isEditColumnVisible()) {
+                addEditColumn(generatedTableColumns);
+            }
+            return generatedTableColumns;
+        },
+        [
+            addEditColumn,
+            enrichColumn,
+            isEditColumnVisible,
+            reorderedTableDefinitionIndexes,
+            selectedColumnsNames,
+        ]
+    );
+
+    const getRows = useCallback(
+        (index) => {
+            if (props.disabled || !props.network) {
+                return [];
+            }
+            const tableDefinition = TABLES_DEFINITION_INDEXES.get(index);
+            const datasourceRows = tableDefinition.getter
+                ? tableDefinition.getter(props.network)
+                : props.network[TABLES_DEFINITION_INDEXES.get(index).resource];
```

Comment:
Why not reuse tableDefinition in stead of TABLES_DEFINITION_INDEXES.get(index) ?

Reply:
Done